### PR TITLE
Bugfix: Fix S3 internalError when try to get an object

### PR DESCRIPTION
**Pull request by vladyslavkurovskyi**
Fix https://github.com/localstack/localstack/issues/6553

### CODE REVIEW

**Comment by vladyslavkurovskyi** [comment]
I have read the CLA Document and I hereby sign the CLA

File: localstack/services/s3/s3_listener.py [hunk 1]
```diff
@@ -688,7 +688,7 @@ def add_accept_range_header(response):
 def is_object_expired(bucket_name: str, key: str) -> bool:
     bucket = BackendState.get_bucket(bucket_name)
     key_obj = bucket.keys.get(key)
-    if not key_obj or not key_obj._expiry:
+    if not key_obj or not getattr(key_obj, "_expiry", False):
```
Comment (thrau) [inline localstack/services/s3/s3_listener.py:691]: if the attribute doesn't exist anymore, then probably it doesn't make sense to try and access it in the first place, even with `getattr`. maybe the attribute moved somewhere else in moto?

**Review by thrau — COMMENTED** [review]
hi @vladyslavkurovskyi , thanks for contributing!

i think our assumption in the method is wrong. there's a method in the new s3 provider `is_key_expired` and `is_object_expired` that @bentsku wrote, maybe he can help port it correctly so we can fix the old one for now?

having a test to catch this would also be good, guess we have it in the new one?

File: localstack/services/s3/s3_listener.py (hunk 1 above)
Comment (vladyslavkurovskyi) [reply localstack/services/s3/s3_listener.py:691]: Looks like you already have similar logic here https://github.com/localstack/localstack/blob/1bbc07609287f0df1310b66e73efb42524ed9d65/localstack/services/s3/utils.py#L118-L121
I guess that it is needed only for `FakeDeleteMarker` instance.

Comment (bentsku) [reply localstack/services/s3/s3_listener.py:691]: Right. The issue sits right there: when getting an object in a versioned bucket, a deleted key would not be deleted but be marked as a `FakeDeleteMarker`, which cannot expire and does not have the `_expiry` attribute.
You can either use the `is_key_expired` helper function here to replace the part you modified, or implement the same logic in your PR.

Comment (bentsku) [reply localstack/services/s3/s3_listener.py:691]: Also, once fixed, you could test the change with the following test:
`tests.integration.s3.test_s3.TestS3.test_get_object_after_deleted_in_versioned_bucket`
You can remove the `xfail` as it should be fixed in both S3 providers.

Comment (vladyslavkurovskyi) [reply localstack/services/s3/s3_listener.py]: @bentsku could you please help me with understanding why `test_get_object_after_deleted_in_versioned_bucket` test [failed](https://app.circleci.com/pipelines/github/localstack/localstack/10947/workflows/73abe8ef-82a5-4461-b3b8-d2e31ed00e48/jobs/74695/tests) and how I can fix this?

File: tests/integration/s3/test_s3.py [hunk 2]
```diff
@@ -856,10 +856,6 @@ def test_head_object_fields(self, s3_client, s3_bucket, snapshot):
         snapshot.match("head-object", response)
 
     @pytest.mark.aws_validated
```
Comment (bentsku) [inline tests/integration/s3/test_s3.py:858]:
```suggestion
    @pytest.mark.aws_validated
    @pytest.mark.skip_snapshot_verify(
        condition=is_old_provider, paths=["$..ContentLanguage", "$..Error.RequestID"]
    )
```

File: localstack/services/s3/s3_listener.py (hunk 1 above)
Comment (bentsku) [reply localstack/services/s3/s3_listener.py]: Oh sorry, I overlooked that part. The current S3 provider has some parity issues with AWS. I suggested a change, you might need to copy/paste it and run the linter again to be sure.
This change allows the snapshot verification to skip those 2 values that are returned by LocalStack but should not be there. Again, sorry for overlooking that part!
And thanks a lot again!
(If you're interested in the snapshot mechanism and how we use it in LocalStack, you can read more here: https://localstack.cloud/blog/2022-08-04-parity-explained/)

Comment (vladyslavkurovskyi) [reply localstack/services/s3/s3_listener.py]: Thank you!

**Review by bentsku — APPROVED** [review]
Awesome! LGTM 👍 thanks a lot for taking the time to fix this bug, greatly appreciated.

**Pull request merged**